openedx/features/enterprise_support/api.py

```diff
@@ -597,102 +582,111 @@ def consent_needed_for_course(request, user, course_id, enrollment_exists=False)
     Wrap the enterprise app check to determine if the user needs to grant
     data sharing permissions before accessing a course.
     """
-    LOGGER.info(
-        "Determining if user [{username}] must consent to data sharing for course [{course_id}]".format(
-            username=user.username,
-            course_id=course_id
-        )
-    )
-
+    # import pdb ; pdb.set_trace()
     enterprise_learner_details = get_enterprise_learner_data_from_db(user)
-    enterprise_customer_uuid = None
-    if enterprise_learner_details:
-        enterprise_customer = enterprise_learner_details[0]['enterprise_customer']
-        enterprise_customer_uuid = enterprise_customer['uuid']
-
-    consent_cache_key = get_data_consent_share_cache_key(user.id, course_id, enterprise_customer_uuid)
-    data_sharing_consent_needed_cache = TieredCache.get_cached_response(consent_cache_key)
-    if data_sharing_consent_needed_cache.is_found and data_sharing_consent_needed_cache.value == 0:
-        LOGGER.info(
-            "Consent from user [{username}] is not needed for course [{course_id}]. The DSC cache was checked,"
-            " and the value was 0.".format(
-                username=user.username,
-                course_id=course_id
-            )
-        )
+    # first enterprise customer in `enterprise_learner_details` is active enterprise and we only have to deal with it
+    active_enterprise_customer = enterprise_learner_details and enterprise_learner_details[0]['enterprise_customer']
```

Comment:
question: Do we only need to handle the active enterprise customer here? There is no way user would try to access courses for enterprise customers who is not active?